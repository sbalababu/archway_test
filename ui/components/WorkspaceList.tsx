import * as React from 'react';
import { List, Table, Row, Col } from 'antd';

import { workspaceColumns, WorkspaceListItem, ListCardToggle } from './';
import { WorkspaceSearchResult } from '../models/Workspace';

interface Props {
  workspaceList: WorkspaceSearchResult[];
  listingMode: string;
  emptyText: string;
  fetching: boolean;

  setListingMode: (mode: string) => void;
  openWorkspace: (id: number) => void;
}

const WorkspaceList = ({ workspaceList, listingMode, emptyText, fetching, setListingMode, openWorkspace }: Props) => {
  return (
    <React.Fragment>
      <ListCardToggle style={{ margin: '12px 0' }} selectedMode={listingMode} onSelect={mode => setListingMode(mode)} />
      {listingMode === 'cards' && (
        <Row style={{ marginTop: 12 }}>
          <Col span={24}>
            <List
              grid={{ gutter: 12, md: 1, lg: 2 }}
              locale={{ emptyText }}
              loading={fetching}
              dataSource={workspaceList}
              renderItem={(workspace: WorkspaceSearchResult) => (
                <WorkspaceListItem workspace={workspace} onSelected={openWorkspace} />
              )}
            />
          </Col>
        </Row>
      )}
      {listingMode === 'list' && (
        <Table
          columns={workspaceColumns}
          dataSource={workspaceList}
          onRow={record => ({
            onClick: () => openWorkspace(record.id),
          })}
        />
      )}
    </React.Fragment>
  );
};

export default WorkspaceList;
