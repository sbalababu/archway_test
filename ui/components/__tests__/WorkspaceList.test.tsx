import * as React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import WorkspaceList from '../WorkspaceList';

describe('WorkspaceList', () => {
  it('it renders correctly', () => {
    const props = {
      workspaceList: [
        {
          id: 1,
          name: 'workspace',
          summary: 'summary',
          status: 'pending',
          behavior: 'simple',
          date_requested: new Date(0),
          date_fully_approved: new Date(0),
          total_disk_allocated_in_gb: 1024,
          total_max_cores: 4,
          total_max_memory_in_gb: 1024,
        },
      ],
      listingMode: 'card',
      emptyText: 'empty',
      fetching: true,
      setListingMode: () => null,
      openWorkspace: () => null,
    };
    const wrapper = shallow(<WorkspaceList {...props} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
