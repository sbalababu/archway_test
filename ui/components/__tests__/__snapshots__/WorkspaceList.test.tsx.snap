// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkspaceList it renders correctly 1`] = `
<Fragment>
  <ListCardToggle
    onSelect={[Function]}
    selectedMode="card"
    style={
      Object {
        "margin": "12px 0",
      }
    }
  />
</Fragment>
`;
